src/main/rx/cc2500_frsky.c

```diff
@@ -0,0 +1,917 @@
+            timeOfLastHopUs = timeNowUs;
+        }
+        break;
+    }
+    return ret;
+}
+
+void frsky_show_partinfo(void)
+{
+    uint8_t partnum, version;
+    //start idle
+    cc25xx_strobe(RFST_SIDLE);
+
+    //check version:
+    debug("frsky: cc25xx partnum 0x");
+    partnum = cc25xx_get_register_burst(PARTNUM);
+    debug_put_hex8(partnum);
+
+    debug(" version 0x");
+    version = cc25xx_get_register_burst(VERSION);
+    debug_put_hex8(version);
+    debug_put_newline();
+
+    if (cc25xx_partnum_valid(partnum, version)){
+        debug("frsky: got valid part and version info\n");
+    }else{
+        debug("frsky: got INVALID part and version info?!\n");
+    }
+    debug_flush();
+}
+
+
+static void frsky_configure(void)
+{
+    debug("frsky: configure\n"); debug_flush();
+
+    //start idle
+    cc25xx_strobe(RFST_SIDLE);
+    
+    // IOCFG0,1,2 is set in hal code (it is specific to the board used)
+    cc25xx_set_gdo_mode();
+
+    //normal config
+    cc25xx_set_register(MCSM1    ,0x0F); //go back to rx after transmission completed //0x0C = stay idle;
+    cc25xx_set_register(MCSM0    ,0x18);
+    cc25xx_set_register(PKTLEN   ,FRSKY_PACKET_LENGTH); //on 251x this has to be exactly our size
+    cc25xx_set_register(PKTCTRL0 ,0x05);
+    cc25xx_set_register(PA_TABLE0,0xFF);
+    cc25xx_set_register(FSCTRL1  ,0x08); //D4R-II seems to set 0x68 here ?! instead of 0x08
+    cc25xx_set_register(FSCTRL0  ,0x00);
+    //set base freq 2404 mhz
+    cc25xx_set_register(FREQ2    ,0x5C);
+    cc25xx_set_register(FREQ1    ,0x76);
+    cc25xx_set_register(FREQ0    ,0x27);
+    cc25xx_set_register(MDMCFG4  ,0xAA);
+    cc25xx_set_register(MDMCFG3  ,0x39);
+    cc25xx_set_register(MDMCFG2  ,0x11);
+    cc25xx_set_register(MDMCFG1  ,0x23);
+    cc25xx_set_register(MDMCFG0  ,0x7A);
+    cc25xx_set_register(DEVIATN  ,0x42);
+    cc25xx_set_register(FOCCFG   ,0x16);
+    cc25xx_set_register(BSCFG    ,0x6C);
+    cc25xx_set_register(AGCCTRL2 ,0x03);
+    cc25xx_set_register(AGCCTRL1 ,0x40); //D4R uses 46 instead of 0x40);
+    cc25xx_set_register(AGCCTRL0 ,0x91);
+    cc25xx_set_register(FREND1   ,0x56);
+    cc25xx_set_register(FREND0   ,0x10);
+    cc25xx_set_register(FSCAL3   ,0xA9);
+    cc25xx_set_register(FSCAL2   ,0x05);
+    cc25xx_set_register(FSCAL1   ,0x00);
+    cc25xx_set_register(FSCAL0   ,0x11);
+    //???FSTEST   , 0x59);
+    cc25xx_set_register(TEST2    ,0x88);
+    cc25xx_set_register(TEST1    ,0x31);
+    cc25xx_set_register(TEST0    ,0x0B);
+    //???FIFOTHR  = 0x07);
+    cc25xx_set_register(ADDR     ,0x00);
+
+    //for now just append status
+    cc25xx_set_register(PKTCTRL1, CC2500_PKTCTRL1_APPEND_STATUS);
+    debug("frsky: configure done\n"); debug_flush();
+}
+
+
+void wdt_reset(void)
+{
+    IWDG_ReloadCounter();
+}
+
+static uint32_t timeoutMs;
+void timeout_set(uint32_t ms)
+{
+    timeoutMs = millis() + ms;
+}
+
+bool timeout_timed_out(void)
+{
+    return millis() > timeoutMs ? true : false;
```

Comment:
`cmp32(millis(), timeoutMs) > 0`  will handle counter overflow correctly (after 49 days of uptime)